app/utils/providers/webauthn.server.ts

```diff
@@ -0,0 +1,151 @@
+import { type Authenticator } from '@prisma/client'
+import { type WebAuthnOptions, WebAuthnStrategy } from 'remix-auth-webauthn'
+import { prisma } from '../db.server.ts'
+import { type ProviderUser, type AuthProvider } from './provider.ts'
+
+// const shouldMock = process.env.WEB_AUTHN_ORIGIN.startsWith('MOCK_')
+
+export class WebAuthnProvider implements AuthProvider {
+	getAuthStrategy() {
+		return new WebAuthnStrategy<ProviderUser>(
+			{
+				rpName: 'Epic Stack WebAuthn Example',
+				rpID:
+					process.env.NODE_ENV === 'development'
+						? 'localhost'
+						: process.env.WEB_AUTHN_ORIGIN,
+				origin: process.env.WEB_AUTHN_ORIGIN,
```

Comment:
If you make these a function, they will accept the `request` which you can then use to dynamically determine their values based on the request (you can use `getDomainUrl`). Which will eliminate the environment variable 🎉

Reply:
Hello @kentcdodds, i don't get how i will be able to pass the request in order to use the getDomainUrl.
WebAuthnProvider is instanciate along side the GitHubProvider in connections.server.ts in the providers record . 

This record is then use in the auth.server.ts to initialise remix-auth Authenticator with the differents strategies.

From what I understand, this is execute when the server start and not with a resquest to the server. :/

Did I miss something ?

Reply:
```ts
rpID: request => getDomainUrl(request),
origin: request => getDomainUrl(request),
```

Reply:
Oh thanks ! I didn't see that the signature of rpID can accept a string or a Request.

Unfortunately, the signature of origin is :
```ts
     rpID: string | ((request: Request) => Promise<string> | string);
    /**
     * Website URL (or array of URLs) where the registration can occur
     */
    origin: string | string[];
    ```

Reply:
I'm pretty sure that's a mistake. @alexanderson1993?

Reply:
Oversight on my part. I'll push an update today.

Reply:
Thanks @alexanderson1993, if it's ok with you I can send you a Pull Request

Reply:
I've already made the change and pushed v0.0.11. Give it an update and see if it's working for you 👍

Reply:
That's fast, thank you. 
It's working I can remove my new environnement variable ! 👍